### ISSUE TITLE
Rule Change: excludeObjects parameter for no-restricted-properties

### ISSUE THREAD

**Issue by osdiab** [eslint/eslint#19348]
### What rule do you want to change?

excludeObjects parameter for no-restricted-properties

### What change do you want to make?

Generate fewer warnings

### How do you think the change should be implemented?

A new option

### Example code

```js
If the rule is


"no-restricted-properties": ["warn", {
  excludeObjects: "router",
  property: "push",
  message: "Prefer [...previousArray, newValue] because it does not mutate the array in place; if you need in-place mutations, disable this rule and explain why"
}]


then it would behave like so:


myArray.push(1) // this errors

const router = useRouter();
router.push("/abc") // this does not error
```

### What does the rule currently do for this code?

no way to exempt some objects from the rule, which leads to false positives for the above scenario

### What will the rule do after it's changed?

allow exempting objects to allow greater control over code standards

### Participation

- [x] I am willing to submit a pull request to implement this change.

**Comment by Tanujkanti4441** [comment]
Hi @osdiab, thanks for the issue, 

Allowing property for a particular object make sense to me, may be we are restricting any property in our project but a module has the property with the same name or vice versa then it will be a useful feature.

let's have some more opinions from @eslint/eslint-team.

**Comment by nzakas** [comment]
I'm not sure I understand what's being proposed. What exactly does it mean to "exempt objects"? 

The rule already only applies to objects that you specifically name, so if you don't name `router`, then it won't apply.

Please problem a description of the *problem  you're trying to solve*, not the solution you think you want.

**Comment by Tanujkanti4441** [comment]
When we use only `property` option in `no-restricted-properties` it will restrict the specified property for all the objects, so the proposal is to add an option to prevent some objects from being reported in this case.

```js
/* eslint no-restricted-properties: [2, {
    "property": "push"
}] */


myArray.push(1) // this will be reported

const router = useRouter();
router.push("/abc") // this will be reported
```
If we want to report `push` property for all the objects but not for `router` object, right now we don't have any option to do that.

**Comment by Tanujkanti4441** [comment]
@osdiab would you please confirm if description in https://github.com/eslint/eslint/issues/19348#issuecomment-2620934235 is what you proposed or not?

**Comment by nzakas** [comment]
@eslint/eslint-tsc thoughts on this?

**Comment by mdjermanovic** [comment]
I'm fine with adding an option to allow some objects, though I'd prefer `allowObjects` name, and the value should probably be an array of strings rather than a single string. We should also ensure that the new option and the existing `object` option cannot be specified together.
